### PR TITLE
Upgrade sld-parser, make use of SLD 1.1

### PR DESCRIPTION
**Pull request by weskamm**
## Description

This update the geostyler-sld-parser package and adds options to the CodeEditor to select the parser with symbology encoding features.

image

## Related issues or pull requests

Please list issues or pull requests that the changes you propose are related to. It does not matter if they are still open and/or unmerged, any link is appreciated.

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [x] Dependency updates
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [ ] Yes
- [x] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [BSD 2-Clause License](https://github.com/geostyler/geostyler/blob/master/)
- [x] I have followed the [guidelines for contributing](https://github.com/geostyler/geostyler/blob/master/CONTRIBUTING.md)
- [x] The proposed change fits to the content of the [code of conduct](https://github.com/geostyler/geostyler/blob/master/CODE_OF_CONDUCT.md)
- [x] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [x] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

File: src/Component/CodeEditor/CodeEditor.tsx [hunk 1]
```diff
@@ -343,6 +368,25 @@ export class CodeEditor extends React.Component<CodeEditorProps, CodeEditorState
...
+                className="gs-code-editor-format-select"
+                style={{ width: 100 }}
+                onSelect={this.onUnitSelect}
+                value={symbolizerUnit}
+              >
+                <Option value="pixel">{locale.symbolizerUnitsPixel}</Option>
+                <Option value="meter">{locale.symbolizerUnitsMeter}</Option>
+                <Option value="foot">{locale.symbolizerUnitsFoot}</Option>
+              </Select>
+            </>
```
Comment (KaiVolland) [inline src/Component/CodeEditor/CodeEditor.tsx:388]: Please move this to a seperate component.

**Review by KaiVolland — CHANGES_REQUESTED** [review]
I think you might to tweak the code in `getParserOptions` too. So you could add both `SLD 1.0.0` and `SLD 1.1.0` parser.

**Comment by weskamm** [comment]
> I think you might to tweak the code in `getParserOptions` too. So you could add both `SLD 1.0.0` and `SLD 1.1.0` parser.

No need, as its given as prop (`parsers`). Already tested and working on the demo, but cannot create PR yet

File: src/Component/CodeEditor/CodeEditor.tsx (hunk 1 above)
Comment (weskamm) [reply src/Component/CodeEditor/CodeEditor.tsx:388]: a label and a select? are you sure you want me to do that?

Comment (KaiVolland) [reply src/Component/CodeEditor/CodeEditor.tsx:388]: Definetly yes. Its a component by definition so it should have its own file. We have many components with less special handling: compare https://github.com/geostyler/geostyler/blob/master/src/Component/Symbolizer/Field/RotateField/RotateField.tsx

Comment (weskamm) [reply src/Component/CodeEditor/CodeEditor.tsx:388]: well looking at the code above and below i dont get the point. there is another similar select, not a standalone component. and there are 2 buttons, no standalone components

Comment (weskamm) [reply src/Component/CodeEditor/CodeEditor.tsx]: i refactored the select to a standalone component

**Review by KaiVolland — APPROVED** [review]
Nice one.

**Comment by weskamm** [comment]
Sorry for bumping in another commit after review, but that one is a real small one... :-)
Thanks for review!

**Comment by KaiVolland** [comment]
> Sorry for bumping in another commit after review, but that one is a real small one... :-)
> Thanks for review!

So `metre` is correct? :)

**Comment by weskamm** [comment]
Yes, its the only valid definition for meter ... :cry:

File: src/Component/Symbolizer/SLDUnitsSelect/SLDUnitsSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  render() {
+    const {
+      locale
+    } = this.props;
+    const {
+      symbolizerUnit
+    } = this.state;
+
+    return (
+      <>
```
Comment (jansule) [inline src/Component/Symbolizer/SLDUnitsSelect/SLDUnitsSelect.tsx:100]: is this intentional?

**Review by jansule — COMMENTED** [review]
Nice work @weskamm. Just one question

File: src/Component/Symbolizer/SLDUnitsSelect/SLDUnitsSelect.tsx (hunk 2 above)
Comment (weskamm) [reply src/Component/Symbolizer/SLDUnitsSelect/SLDUnitsSelect.tsx:100]: yeah its the needed parent container, slimmed down div ...

**Pull request merged**